### PR TITLE
Auth0 Extend Blog: Why is Serverless Extensibility better than Webhooks?

### PR DESCRIPTION
**Pull request by NotMyself**
In Progress...

screencapture-0-0-0-0-4000-blog-why-is-serverless-extensibility-better-than-webhooks-1505139976777

### CODE REVIEW

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+![Hidden Complexity](https://cdn.auth0.com/website/blog/extend/why-is-serverless-extensibility-better-than-webhooks/hidden-complexity.png)
+
+## Going Beyond Webhooks
+
+A new approach is starting to emerge in the industry, one that we call Serverless Extensibility. Instead of placing the burden on customers, SasS products can leverage a Serverless platform to allow users to author and execute extensions directly in the product.
+
+![Serverless Extensibility](https://cdn.auth0.com/website/blog/extend/why-is-serverless-extensibility-better-than-webhooks/serverless-extensibility.png)
+
+The hook points within the product that used to call out via Webhooks, now call into a Serverless runtime for securely running third party extensions.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: This replaces this line and the next.

Users can now create custom extensions within the product using an inline code editor, choosing from a pre-defined selection. At runtime, The SaaS calls out to a Serverless provider to securely execute the code.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Those extensions can be pre-defined and selected from a menu, or created within the product directly using an inline code editor.
+
+All of this without consideration for hosting, scalability, or monitoring removing as much friction as possible to the user extending the product. Serverless Extensibility makes it much easier and faster to customize your product and get value to your users. Not only that, but it saves your customers money as they greatly reduce up-front development cost, and they no longer have to deal with ongoing hosting and maintenance. Removing that friction and cost means higher user retention, such as the 10x retention we’ve seen within our own identity product.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Serverless Extensibility makes it much easier and faster to customize your product and get the value to your users. Not only that, but it saves your customers money as they substantially reduce up-front development cost, and they no longer have to deal with ongoing hosting and maintenance. Removing that friction and cost means higher user retention, which our data shows can be as high as 10x.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+## Serverless Extensibility in the Wild
+
+### Auth0
+
+Here at Auth0, we have been working on extensibility for a while. We are known for offering the simplest set of tools to solve the most complex identity use cases. With Auth0 Identity, you can easily authenticate and authorize apps and APIs with any identity provider running on any stack, device, or cloud.
+
+For several years customers have been able to execute arbitrary Node.js code during the authorization process using [Rules](https://auth0.com/docs/rules/current). Several templates are available or create an empty Rule to customize right in the management dashboard.
+
+Recently, [Hooks](https://auth0.com/docs/hooks) was released based on the Auth0 Extend serverless platform that allows you to customize the behavior of Auth0 at credential exchange, pre or post user registration extension points.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Recently, we added hooks, allowing users to customize the behavior of Auth0 at credential exchange, or at pre and post user registration extension points.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+![Auth0 Hook Editor](https://cdn.auth0.com/website/blog/extend/why-is-serverless-extensibility-better-than-webhooks/hooks-editor2.png)
+
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Add as a summary

"Using Auth0's extensibility has enabled our customers to easily customize the product to handle custom business logic and integrate with a variety of 3rd party services."

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+Twilio recently augmented their Webhook based extensibility with [Twilio Functions](https://www.twilio.com/blog/2017/05/introducing-twilio-functions.html). In the introduction presentation at Signal 2017, Carter Rabasa explained the reasoning this way.
+
+> **"The first thing that Twilio tells you to do is download a web application. And then we ask you to go deploy that web application to some server, somewhere."**<br />
+> [Carter Rabasa](https://www.youtube.com/watch?v=PKodXk6L5Eo) - Product Manager @ Twilio
+
+Now you can easily create a function that handles incoming voice calls or messages right in the Twilio console. When saved, the function is deployed to a serverless environment and can now be attached to incoming events.
+
+![Twillio Editor](https://cdn.auth0.com/website/blog/extend/why-is-serverless-extensibility-better-than-webhooks/twilio-editor.png)
+
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Add

"Using Twilio Functions enables customers to now on board much more quickly to using Twilio's platform, than was previously possible."

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+### Stamplay
+
+[Stamplay](https://stamplay.com/) is a business automation platform that provides the ability to connects any application, database or service regardless of if data is sitting in the cloud or behind a tightly secure firewall on-premises.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Stamplay is a business automation platform that provides the ability to connect any application, database or service regardless of if data is sitting in the cloud or on-premises.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+With Stamplay you can plug in a large and growing [list of integrations](https://stamplay.com/product#integrations) and use them to orchestrate communication between systems based on event triggers. It offers a beautiful point and click task editor that allows any business professional to wire together interactions.
+
+In January of 2016, Stamplay released Code Blocks a built in extensibility point that solved a specific need in their product. Giuliano Lacobelli introduced it this way.
+
+> **"Sometimes you cannot get around the need for some custom code."**<br />
+> [Giuliano Lacobelli](https://blog.stamplay.com/code-blocks-101-d4741b107ec0) - Co-Founder @ Stamplay
+
+Now Stamplay customers can create any integration not currently offered by the platform directly and consume it directly in the point and click editor.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Now Stamplay customers can create custom integrations within the inline editor, and use them directly on Stamplay's drag and drop environment.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+![Stamplay Editor](https://cdn.auth0.com/website/blog/extend/why-is-serverless-extensibility-better-than-webhooks/stamplay-editor2.png)
+
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Add

"Using Stamplay Blocks allows customers to go well beyond the out-of-the-box integrations, providing a stickier experience."

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+## Serverless Extensibility in your SaaS
+
+With over three years of experience implementing serverless extensibility in our identity products, [Auth0 Extend](https://auth0.com/extend/) offers this capability to any SaaS looking to remove the Webhook burden from their end users.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Implementing Serverless Extensibility is hard. We know as we have been building out our own solution which we run at scale, globally, for 3 years. Fortunately, with Auth0 Extend, you can do it much more easily.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+Auth0 Extend is an Extensibility as a Service platform that makes it easy for your customers, sales engineers, and partners to quickly extend your SaaS. It offers an embeddable, customizable code editor and a cloud-based service for running extensions on demand, securely, and at scale.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Auth0 Extend is a managed extensibility platform that you can plug into your SaaS. It makes it easy for your customers, sales engineers, and partners to quickly extend your SaaS. Extend includes an embeddable, customizable code editor and a cloud-based service for running extensions on demand, securely, and at scale.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+You can [create a trial account](https://auth0.com/extend/try) today that allows you to explore the capabilities at no cost.
+
+## Summary
+
+Webhooks are a clean and clear way to add extensibility points into your application, but they put the burden on your customers to complete the circle that makes them useful.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Webhooks are a clean and easy way for you to add extensibility points into your SaaS, but they put an extra on-going burden on your customers, making them less useful.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+You can [create a trial account](https://auth0.com/extend/try) today that allows you to explore the capabilities at no cost.
+
+## Summary
+
+Webhooks are a clean and clear way to add extensibility points into your application, but they put the burden on your customers to complete the circle that makes them useful. 
+
+Serverless Extensibility removes as much friction as possible, allowing customers to customize usage of your platform in a secure, scalable, and reliable way that is offered directly in the product.
+
+Removing this friction means customers can focus on their core business and move quickly to deliver value to the market.
```
Comment (glennblock) [inline _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Removing this friction makes customers stickier to your product, allowing them to focus on their core business and more quickly to deliver value to the market.

File: _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown (hunk 5 above)
Comment (NotMyself) [reply _posts/2017-09-11-why-is-serverless-extensibility-better-than-webhooks.markdown]: Done

**Pull request merged**